### PR TITLE
permissions iteration 1

### PR DESCRIPTION
**Pull request by corp-0**
### Purpose
Adds the new permissions system. It will allow server owners and administrators to set arbirtrary ranks in a config file and grant permissions to said ranks.

In this iteration, this is still not integrated into the game, but I wanted to PR as soon as possible because I don't know when will I have time to complete it.

### Notes
Since this is still not visible by players, I'm not including a changelog record.

### CODE REVIEW

**Comment by corp-0** [comment]
Don't merge yet, I have to make some changes in the build script

**Comment by corp-0** [comment]
done. Let's see if this works for tests.

File: UnityProject/Assets/StreamingAssets/admin/permissions.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+identifier = "hostPlayer"
+rank = "host"
+
+[[players]]
+identifier = "adminPlayer"
+rank = "admin"
+
+[[players]]
+identifier = "normalPlayer"
+rank = "normal"
```
Comment (ThatDan123) [inline UnityProject/Assets/StreamingAssets/admin/permissions.toml]: Maybe add an example comment of how to add a user to one of these?

Or is this just a file to setup the ranks and perms and then in another you'll give users that rank?

Comment (corp-0) [reply UnityProject/Assets/StreamingAssets/admin/permissions.toml]: In this file you create the ranks and assign them to users.

To add a new record you just add the [[players]] list and under it, the identifier (ugly ass firebase string) and the rank you created above.

I thought it was self explanatory.

Comment (ThatDan123) [reply UnityProject/Assets/StreamingAssets/admin/permissions.toml]:
```suggestion
[[players]]
identifier = "hostPlayerUserID"
rank = "host"

[[players]]
identifier = "adminPlayerUserID"
rank = "admin"

[[players]]
identifier = "normalPlayerUserID"
rank = "normal"
```

Maybe that then?

Comment (corp-0) [reply UnityProject/Assets/StreamingAssets/admin/permissions.toml]: Sure, the ranks I created there are merely for testing purposes. I didn't intend to do the actual ranks we will have nor the permissions.

The normal user on the bottom only represents a record without permissions. Of course we don't need it in the game.

Comment (corp-0) [reply UnityProject/Assets/StreamingAssets/admin/permissions.toml]: I separated the example config file form the test one. Now the example is a better representation of reality. Take a look.

**Review by ThatDan123 — APPROVED** [review]

File: UnityProject/Assets/StreamingAssets/admin/permissions.toml (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/StreamingAssets/admin/permissions.toml]: Yeah, i think that explains it well.

**Pull request merged**